src/core/shared_state/currentMapPosition.ts

```diff
@@ -36,43 +36,37 @@ function jumpTo(map: Map, position: CenterZoomPosition) {
   }, 100);
 }
 
-// TODO: #20160 update currentMapPositionAtom to reatom v3
-export const currentMapPositionAtom = createAtom(
-  {
-    setCurrentMapPosition: (mapPosition: CenterZoomPosition) => mapPosition,
-    setCurrentMapBbox: (mapBbox: Bbox | [[number, number], [number, number]]) => mapBbox,
-    updateCurrentMapPosition: (mapPosition: CenterZoomPosition) => mapPosition,
-    currentMapAtom,
-  },
-  ({ onAction }, state: CurrentMapPositionAtomState = null) => {
-    onAction('setCurrentMapPosition', (position) => {
-      const map = currentMapAtom.getState();
-      if (map) {
-        jumpTo(map, position);
-      }
-      state = position;
-    });
-
-    onAction('setCurrentMapBbox', (bbox) => {
-      let position = { bbox: bbox.flat() } as MapPosition;
-      const map = currentMapAtom.getState();
-      if (map) {
-        const camera = getCameraForBbox(bbox, map);
-        if (camera.center && 'lng' in camera.center) {
-          const { zoom } = camera;
-          const { lat, lng } = camera.center;
-          position = { ...position, lat, lng, zoom: zoom ?? map.getZoom() };
-          jumpTo(map, position);
-        }
-      }
-      state = position;
-    });
+export const currentMapPositionAtom = atom<CurrentMapPositionAtomState>(
+  null,
+  '[Shared state] currentMapPositionAtom',
+);
 
-    onAction('updateCurrentMapPosition', (position) => {
-      state = position;
-    });
+export const setCurrentMapPosition = action((ctx, position: CenterZoomPosition) => {
+  const map = currentMapAtom.getState();
+  if (map) {
+    jumpTo(map, position);
+  }
+  currentMapPositionAtom(ctx, position);
+}, 'setCurrentMapPosition');
 
-    return state;
+export const setCurrentMapBbox = action(
+  (ctx, bbox: Bbox | [[number, number], [number, number]]) => {
+    let position = { bbox: bbox.flat() } as MapPosition;
+    const map = currentMapAtom.getState();
+    if (map) {
+      const camera = getCameraForBbox(bbox, map);
+      if (camera.center && 'lng' in camera.center) {
+        const { zoom } = camera;
+        const { lat, lng } = camera.center;
+        position = { ...position, lat, lng, zoom: zoom ?? map.getZoom() };
+        jumpTo(map, position);
+      }
+    }
+    currentMapPositionAtom(ctx, position);
   },
-  '[Shared state] currentMapPositionAtom',
+  'setCurrentMapBbox',
 );
+
+export const updateCurrentMapPosition = action((ctx, position: CenterZoomPosition) => {
```

Comment:
looks like this action not needed as we can use directly currentMapPositionAtom(ctx, position); in React hook as well

Reply:
Fixed, removed the action